onchain/rollups/contracts/library/LibOutputValidation.sol

```diff
@@ -5,10 +5,102 @@ pragma solidity ^0.8.8;
 import {MerkleV2} from "@cartesi/util/contracts/MerkleV2.sol";
-import {OutputEncoding} from "../common/OutputEncoding.sol";
 import {OutputValidityProof} from "../common/OutputValidityProof.sol";
 
 /// @title Output Validation Library
+///
+/// @notice The diagram below aims to better illustrate the algorithm
+/// behind calculating the epoch hash. Each component in the diagram is
+/// labeled, so it can be more easily referenced in this documentation.
+/// The diagram is laid out in a top-down fashion, with the epoch hash
+/// being the top-most component, but the text will traverse the diagram
+/// in a bottom-up fashion.
+///
+/// The off-chain machine may, while processing an input, generate
+/// zero or more outputs. For the scope of this algorithm, let us assume
+/// that an output is an arbitrary array of bytes.
+///
+/// Every output is hashed into a 256-bit word (A), which is then
+/// divided into four 64-bit words. From these words, a Merkle tree
+/// is constructed from the bottom up (B). The result is a Merkle
+/// root hash (C) that fully represents the contents of the output.
+///
+/// Now, this process is repeated for every output generated by an input.
+/// The Merkle root hashes (C) derived from these outputs are then ordered
+/// from oldest to newest (D), and used to construct yet another Merkle tree
+/// (E). The result is a Merkle root hash (F) that fully represents the
+/// contents of every output generated by an input.
+///
+/// We once again repeat this process, but now for every input accepted in a
+/// certain epoch. Then, we organize the Merkle root hashes (F) calculated
+/// in the previous step and order them from oldest to newest (G). From these
+/// Merkle root hashes (F), we build a final Merkle tree (H). The result is
+/// a Merkle root hash (I) that fully represents the contents of every output
+/// generated by every input in a certain epoch.
+///
+/// Finally, this Merkle root hash (I) is combined with the machine state
+/// hash (J) to obtain the epoch hash (K).
+///
+/// ```
+///                     ┌──────────────┐
+///           ┌─────────┤Epoch Hash (K)├────────┐
+///           │         └──────────────┘        │
+///           │                                 │
+///           │                                 │
+///           │                      ┌──────────▼───────────┐
+///           │                      │Machine State Hash (J)│
+///           │                      └──────────────────────┘
+///     ┌─────▼─────┐
+///     │Merkle Root│ ───> Epoch's output hashes root hash (I)
+///     └───────────┘
+///           x
+///          xxx         │
+///         xxxxx        │
+///        xxxxxxx       │
+///       xxxxxxxxx      ├──> Epoch's outputs Merkle tree (H)
+///      xxxxxxxxxxx     │
+///     xxxxxxxxxxxxx    │
+///    xxxxxxxxxxxxxxx   │
+///   xxxxxxxxxxxxxxxxx
+/// ┌────────┬─┬────────┐
+/// │   ...  │┼│  ...   │ ───> For each input in the epoch (G)
```

Comment:
Before rebase, `(G)` appears again in the struct `OutputValidityProof`'s comments, i.e.
```
/// @param inputIndexWithinEpoch Which input, inside the epoch, the output belongs to (G)
```
But because the struct `OutputValidityProof` is moved to a new file after rebase, this reference is also removed